### PR TITLE
Porting remaining Changes from Write component Library into Persistence component

### PR DESCRIPTION
**Pull request by prasar-ashutosh**
#### What type of PR is this?
Feature.

#### What does this PR do / why is it needed ?
We are introducing a common library to handle batch ingestion logic with an initial focus on batch milestoning into relational databases. This library consolidates the implementation across multiple backend persistence implementations. It also provides an in-memory implementation that will be of use for fast testing.
This PR is for porting the remaining Changes from Write component Library into Persistence component

#### Which issue(s) this PR fixes:

Fixes #
None. But this is the first of a series of PRs addressing https://github.com/finos/legend-engine/issues/920

#### Does this PR introduce a user-facing change?
No

### CODE REVIEW

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/AppendOnlyPlanner.java [hunk 1]
```diff
@@ -62,6 +62,24 @@ class AppendOnlyPlanner extends Planner
     {
         super(datasets, ingestMode, plannerOptions);
 
+        // validate
```
Comment (rafaelbey) [inline legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/AppendOnlyPlanner.java:65]: Given we already following a visitor pattern, can we use that patter to simplify this logic? 

```java
ingestMode.deduplicationStrategy().accept(new DeduplicationStrategyVisitor<Void>()
    {
        @Override
        public Boolean visitAllowDuplicates(AllowDuplicatesAbstract allowDuplicates)
        {
            validatePrimaryKeysIsEmpty(primaryKeys);
            return null;
        }

        @Override
        public Boolean visitFilterDuplicates(FilterDuplicatesAbstract allowDuplicates)
        {
            validatePrimaryKeysNotEmpty(primaryKeys);
            return null;
        }

        @Override
        public Boolean visitFailOnDuplicates(FailOnDuplicatesAbstract allowDuplicates)
        {
            validatePrimaryKeysNotEmpty(primaryKeys);
            return null;
        }
    })
```

This has the benefit of better informing developers of the complexities of adding a new dedup strategy, given the visitor will highlight at compile time that this particular validation will need to change too.

Note that I will suggest to also delete `DeduplicationStrategyVisitors` given seems unnecessary if visitor actually fits.

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/BitemporalDeltaPlanner.java [hunk 2]
```diff
@@ -93,10 +97,12 @@ class BitemporalDeltaPlanner extends BitemporalPlanner
...
     private List<FieldValue> dataFields;
 
-    private final String startAlias = "start_date";
-    private final String endAlias = "end_date";
-    private final String xAlias = "x";
-    private final String yAlias = "y";
+    private static final String START_DATE = "start_date";
+    private static final String END_DATE = "end_date";
+    private static final String X_ALIAS = "x";
+    private static final String Y_ALIAS = "y";
```
Comment (rafaelbey) [inline legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/BitemporalDeltaPlanner.java:103]: Given we are defining this as constant, can we provide more descriptive names.  Ideally, the purpose/intention of these could be encapsulated on the name, helping self-documenting the code.

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/GeneratorResultAbstract.java [hunk 3]
```diff
@@ -38,7 +38,7 @@
 )
 public abstract class GeneratorResultAbstract
 {
-    private static final String SINGLE_QUOTE = "'";
+    public static final String SINGLE_QUOTE = "'";
```
Comment (rafaelbey) [inline legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/GeneratorResultAbstract.java:41]: Seems odd to define a general-purpose constant on this class.  Even further, do we need a constant for this well know value?

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/AppendOnlyPlanner.java (hunk 1 above)
Comment (davidsunglee) [reply legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/AppendOnlyPlanner.java:65]: Agreed, Putting together follow-up PR to address this now

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/GeneratorResultAbstract.java (hunk 3 above)
Comment (davidsunglee) [reply legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/GeneratorResultAbstract.java:41]: Will find usage and cleanup in forthcoming PR

**Review by rafaelbey — APPROVED** [review]
A mentioned by @davidsunglee , feedback items will be addressed on incoming new PR.  Approving and merging this one.

**Pull request merged**